whatToDad_site/whatToDad_app/views.py

```diff
@@ -15,6 +15,38 @@ def get(self,request):
             template_name = 'post_detail.html'
         )
 
+class PostAction(View):
+    def get(self, request, post_id):
+        post = Post.objects.get(id=post_id)
+        post_form = PostForm(instance=post)
+
+        html_data = {
+            'post':post,
+            'form':post_form
+        }
+
+        return render(
+            request,
+            template_name = 'post_action.html',
+            context= html_data
+        )
+
+    def post(self, request, post_id):
+        post = Post.objects.get(id=post_id)
+
+        if 'delete' in request.POST:
+            post.delete()
```

Comment:
Minor thing but you don't need to load the whole post. You can just delete it by id

https://stackoverflow.com/questions/3805958/how-to-delete-a-record-in-django-models